tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,232 @@
+        channel (int) : logging channel
+
+    Returns:
+        bool : Returns True if version matches, False otherwise
+
+    """
+
+    cluster_logging_csv, elasticsearch_csv, images = get_csv_version(channel)
+    if config.UPGRADE['upgrade_logging_channel'] in (
+        cluster_logging_csv and elasticsearch_csv
+    ):
+        logger.info(
+            f"Upgraded version of Cluster-logging "
+            f"operator {cluster_logging_csv}"
+        )
+        logger.info(
+            f"Upgraded version of Elastic-search "
+            f"operator {elasticsearch_csv}"
+        )
+        return True
+    return False
+
+
+@retry(AttributeError, 5, 30, 2)
+def check_csv_logging_phase(channel):
+    """
+    The function checks in the CSV for the phase "Succeeded"
+    for cluster-logging operator and Elastic-search
+    operator
+
+    Args:
+        channel (int) : Logging channel
+
+    Returns:
+        bool: True if CSV status is Succeeded, False otherwise
+
+    """
+
+    cluster_version_csv, elasticsearch_csv, images = get_csv_version(channel)
+    logging_csv = CSV(
+        resource_name=cluster_version_csv,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    clo_phase = logging_csv.get().get('status').get('phase')
+    logging_csv.get().get('status').get('phase')
+
+    es_csv = CSV(
+        resource_name=elasticsearch_csv,
+        namespace=constants.OPENSHIFT_OPERATORS_REDHAT_NAMESPACE
+    )
+    eso_phase = es_csv.get().get('status').get('phase')
+
+    if clo_phase and eso_phase == 'Succeeded':
+        logger.info("CSV phase Succeeded")
+        return True
+
+    logger.info("Upgrade not completed yet")
+    return False
+
+
+def upgrade_info(channel):
+    """
+     Function to provide CSV names and images pre and post upgrade
+
+     Args:
+         channel (int): Logging channel
+
+    """
+
+    logging_csv, elasticsearch_csv, images = get_csv_version(channel)
+    logger.info(
+        f"Cluster-logging CSV for channel {channel} "
+        f"{logging_csv}"
+    )
+    logger.info(
+        f"Elastic-search CSV for channel {channel} "
+        f"{elasticsearch_csv}"
+    )
+    logger.info(
+        f"Images of cluster-logging components for channel {channel} "
+        f"{images}"
+    )
+
+
+@post_ocp_upgrade
+@pytest.mark.polarion_id("OCS-2201")
+class TestUpgradeLoggingAfterOCPUpgrade():
+    """
+    This class contains test for upgrade openshift-logging
+    after OCP upgrade
+    1. Monitors OCS health
+    2. Checks OCP version
+    3. Checks logging version, if mismatched then upgrades logging
+    4. Checks Elasticsearch and clusterlogging operator
+    5. Checks logging health post upgrade
+
+    """
+
+    def test_upgrade_logging(self):
+        """
+        This function contains test to upgrade openshift-logging
+        with Entry and Exit criteria for checks
+
+        """
+
+        ceph_cluster = CephCluster()
+        with CephHealthMonitor(ceph_cluster):
+
+            #  Pre-check
```

Comment:
you may want to add some logging information here, if test failed in this stage it would me easier to debug, something like:
```logger.info("Checking cluster logging before starting to upgrade")```